src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshots.java

```diff
@@ -70,17 +123,114 @@ public FileInfo findPhysicalIndexFile(String physicalName) {
      * @return file info or null if file is not present in any of snapshots
      */
     public FileInfo findNameFile(String name) {
-        for (BlobStoreIndexShardSnapshot snapshot : shardSnapshots) {
-            FileInfo fileInfo = snapshot.findNameFile(name);
-            if (fileInfo != null) {
-                return fileInfo;
-            }
-        }
-        return null;
+        return files.get(name);
     }
 
     @Override
-    public Iterator<BlobStoreIndexShardSnapshot> iterator() {
+    public Iterator<SnapshotFiles> iterator() {
         return shardSnapshots.iterator();
     }
+
+    static final class Fields {
+        static final XContentBuilderString FILES = new XContentBuilderString("files");
+        static final XContentBuilderString SNAPSHOTS = new XContentBuilderString("snapshots");
+    }
+
+    static final class ParseFields {
+        static final ParseField FILES = new ParseField("files");
+        static final ParseField SNAPSHOTS = new ParseField("snapshots");
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.startArray(Fields.FILES);
+        for (Map.Entry<String, FileInfo> entry : files.entrySet()) {
+            FileInfo.toXContent(entry.getValue(), builder, params);
+        }
+        builder.endArray();
+
+        builder.startObject(Fields.SNAPSHOTS);
+        for (SnapshotFiles snapshot : shardSnapshots) {
+            builder.startObject(snapshot.snapshot(), XContentBuilder.FieldCaseConversion.NONE);
+            builder.startArray(Fields.FILES);
+            for (FileInfo fileInfo : snapshot.indexFiles()) {
+                builder.value(fileInfo.name());
+            }
+            builder.endArray();
+            builder.endObject();
+        }
+        builder.endObject();
+
+        builder.endObject();
+        return builder;
+    }
+
+    public static BlobStoreIndexShardSnapshots fromXContent(XContentParser parser) throws IOException {
```

Comment:
I was able to puzzle out the format this is reading, but can you add a javadoc comment (or a comment wherever) with just a quick snippet of the JSON this is reading? I think it'd help anyone maintaining it in the future.